src/uploadStrategy/strategy.ts

```diff
@@ -7,5 +7,12 @@ export interface UploadStrategy {
    *
    * @param {BatchUpdateDocuments} batch
    */
-  uploadBatch: (batch: BatchUpdateDocuments) => Promise<AxiosResponse>;
+  upload: (batch: BatchUpdateDocuments) => Promise<AxiosResponse>;
+
+  /**
+   * Async operation to run once the batch upload is complete
+   *
+   * @memberof UploadStrategy
+   */
+  postUpload: () => Promise<void>;
```

Comment:
I see that `FileContainerStrategy` does virtually not use it.
If that's the case, I would define this as an optional, and the caller of this should check if there's a postUpload before calling it.